Source/WebCore/page/Location.cpp

```diff
@@ -50,10 +51,7 @@ Location::Location(LocalDOMWindow& window)
 
 inline const URL& Location::url() const
 {
-    if (!frame())
-        return aboutBlankURL();
-
-    const URL& url = frame()->document()->urlForBindings();
+    const URL& url = LocalDOMWindowProperty::window()->document()->urlForBindings();
```

Comment:
Can we use `m_window->document()` here and null check `m_window` instead of `frame()` above?

Reply:
> Can we use `m_window->document()` here and null check `m_window` instead of `frame()` above?

Yes, will make that change and push an update

Reply:
So compiling `m_window->document()` fails with _“error: no member named 'document' in 'WebCore::DOMWindow'”_.

And `LocalDOMWindowProperty::window()->document()` _also_ now fails with _“error: call to non-static member function without an object argument”_ — due to the changes made in 194c42fd694410278bc8a19d998239cfe043667f.

I guess I need to study the 194c42fd694410278bc8a19d998239cfe043667f changes to figure out what would actually work here instead now.
cc @achristensen07

Reply:
Looks like Alex currently doesn't deal property with remote frames for Location::url(), as per the code:
```
const URL& Location::url() const
{
    RefPtr localFrame = dynamicDowncast<LocalFrame>(frame());
    if (!localFrame)
        return aboutBlankURL();
```

As a result, for now, I'd just do:
```
if (is<LocalDOMWindow>(m_domWindow.get())
    return aboutBlankURL();

const URL& url = downcast< LocalDOMWindow>(*m_domWindow).document()->urlForBindings();
```

Reply:
OK I can build successfully with `const URL& url = downcast<LocalDOMWindow>(*m_domWindow).document()->urlForBindings()`, but the build doesn’t pass the _“does not change document's URL (active but not fully active document)”_ test at http://wpt.live/html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url.window.html

Reply:
OK, with `const URL& url = downcast<LocalDOMWindow>(*m_window).document()->urlForBindings()` it passes the http://wpt.live/html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url.window.html test once again — if I also completely (re)remove the preceding null-frame check:

```c++
RefPtr localFrame = dynamicDowncast<LocalFrame>(frame());
if (!localFrame)
    return aboutBlankURL();
```
I believe that check isn’t actually spec-conformant — which is why the patch in this PR branch had also removed the previous (`s/if (!localFrame)/if (!frame())`) formulation of it:

https://github.com/WebKit/WebKit/pull/18026/commits/cfa36aa83397066f2721e40e6a2cf77053619a89#diff-9fb3dfaab8c435ab08296c7a653f8f5386788616337f1fc2004a8c6e5db3e276L53